open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,505 @@
+            validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        } else {
+            // If periodic limits key is missing, return an error
+            log.error(ErrorConstants.MISSING_PERIOD_LIMITS);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                    ErrorConstants.MISSING_PERIOD_LIMITS,
+                    ErrorConstants.PATH_PERIOD_TYPE);
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    public static JSONObject validateParameterDateTime(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        if (controlParameters.containsKey(ConsentExtensionConstants.VALID_TO_DATE_TIME)) {
+
+            // Retrieve the validDateTime from the control parameters
+            Object validateToDateTime = controlParameters.get(ConsentExtensionConstants.VALID_TO_DATE_TIME);
+
+            if (!isValidDateTimeObject(validateToDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_TO_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+
+            // Validate ISO datetime format for validTo
+            if (!isValidISODateTimeFormat((String) validateToDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_TO_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+        }
+
+        if (controlParameters.containsKey(ConsentExtensionConstants.VALID_FROM_DATE_TIME)) {
+
+            // Retrieve the validDateTime from the control parameters
+            Object validateFromDateTime = controlParameters.get(ConsentExtensionConstants.VALID_FROM_DATE_TIME);
+
+
+            if (!isValidDateTimeObject(validateFromDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_FROM_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+
+            // Validate ISO datetime format for validTo
+            if (!isValidISODateTimeFormat((String) validateFromDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_FROM_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+        }
+
+        String validTo = controlParameters.getAsString(ConsentExtensionConstants.VALID_TO_DATE_TIME);
+        String validFrom = controlParameters.getAsString(ConsentExtensionConstants.VALID_FROM_DATE_TIME);
```

Comment:
Shall we remove these unnecessary variable assignments. Directly pass the getAsString value into the OffsetDateTime.parse() method.